compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/ControlFlowInfos.kt

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.cfa
+
+import kotlinx.collections.immutable.PersistentMap
+import kotlinx.collections.immutable.persistentMapOf
+import org.jetbrains.kotlin.contracts.description.EventOccurrencesRange
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.CFGNode
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.EdgeLabel
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.NormalPath
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.TryExpressionExitNode
+import org.jetbrains.kotlin.fir.symbols.FirBasedSymbol
+import org.jetbrains.kotlin.fir.symbols.impl.FirPropertySymbol
+import java.lang.IllegalStateException
+
+// TODO: this can be `sealed` once all of its subtypes are migrated
```

Comment:
I think there is no sense to make it `sealed` for two reasons:
1. It's better to write concrete implementation of `ControlFlowInfo` close to checker which uses it
2. In future there will be compiler plugins which may want to provide additional CFA with their own implementations of info

Reply:
I see. I'll keep this as a separate, stand-alone, abstract class here.